### PR TITLE
Chemical - Improve various aspects

### PR DESCRIPTION
**Pull request by MiszczuZPolski**
**When merged this pull request will:**
- add removeGlobalEventJIP 
- change breathing and handleGasMaskDur to event based

More to come I guess

### IMPORTANT

- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/chemical/functions/fnc_showChemDetector.sqf [hunk 1]
```diff
@@ -50,17 +50,26 @@ private _exposure = _display displayCtrl 18805;
...
 
-    if ((count _gas) > 0) then {
-        private _distance = _unit distance (_gas select 0);
-        private _radius = (_gas select 0) getVariable [QGVAR(gas_radius), 0];
+    _exposure ctrlSetText (_intensity toFixed 2);
 
-        _exposure ctrlSetText ((linearConversion[0, _radius, _distance, 1, 0, true]) toFixed 2);
+    if (_intensity > 0.1) then {
+        private _lastSoundTime = _unit getVariable QGVAR(lastSoundTime);
+
```
Comment (mazinskihenry) [inline addons/chemical/functions/fnc_showChemDetector.sqf:62]:
```suggestion
_unit setVariable [QGVAR(areaIntensity), 0, true];
```

File: addons/chemical/functions/fnc_showChemDetector.sqf [hunk 2]
```diff
@@ -50,17 +50,26 @@ private _exposure = _display displayCtrl 18805;
...
 
-    if ((count _gas) > 0) then {
-        private _distance = _unit distance (_gas select 0);
-        private _radius = (_gas select 0) getVariable [QGVAR(gas_radius), 0];
+    _exposure ctrlSetText (_intensity toFixed 2);
 
-        _exposure ctrlSetText ((linearConversion[0, _radius, _distance, 1, 0, true]) toFixed 2);
+    if (_intensity > 0.1) then {
+        private _lastSoundTime = _unit getVariable QGVAR(lastSoundTime);
+_unit setVariable [QGVAR(areaIntensity), 0, true];
```
Comment (mazinskihenry) [inline addons/chemical/functions/fnc_showChemDetector.sqf]:
```suggestion
        _unit setVariable [QGVAR(areaIntensity), 0, true];

```

**Comment by mazinskihenry** [comment]
Review done. Everything seems to be in order

File: addons/chemical/CfgVehicles.hpp [hunk 3]
```diff
@@ -19,19 +19,21 @@ class CfgVehicles {
                     showDisabled = 0;
                     icon = QPATHTOF(ui\gasmaskfilter_ca.paa);
                 };
-                class KAT_MuteChemDetector {
-                    displayName = CSTRING(ChemDetector_Mute);
-                    condition = QUOTE([ARR_2(_player,true)] call FUNC(getChemDetectorState));
-                    statement = QUOTE(_player setVariable [ARR_3(QQGVAR(chemDetectorState),false,true)]);
+                class KAT_PutOnGasMask {
+                    displayName = "Put on Gas Mask";
```
Comment (mazinskihenry) [inline addons/chemical/CfgVehicles.hpp]: Stringtable

File: addons/chemical/CfgVehicles.hpp [hunk 4]
```diff
@@ -19,19 +19,21 @@ class CfgVehicles {
...
                     showDisabled = 0;
-                    icon = QPATHTOF(ui\ChemDetectorIcon.paa);
+                    icon = QPATHTOF(ui\Gasmask_icon.paa);
                 };
-                class KAT_UnmuteChemDetector {
-                    displayName = CSTRING(ChemDetector_Unmute);
-                    condition = QUOTE([ARR_2(_player,false)] call FUNC(getChemDetectorState));
-                    statement = QUOTE(_player setVariable [ARR_3(QQGVAR(chemDetectorState),true,true)]);
+                class KAT_TakeOffGasMask {
+                    displayName = "Take off Gas Mask";
```
Comment (mazinskihenry) [inline addons/chemical/CfgVehicles.hpp]: Stringtable

File: .hemtt/launch.toml
Comment (LinkIsGrim) [inline .hemtt/launch.toml:1]: Need this?

File: addons/feedback/script_component.hpp [hunk 5]
```diff
@@ -3,7 +3,7 @@
 #include "\x\kat\addons\main\script_mod.hpp"
 
 // #define DEBUG_MODE_FULL
-// #define DISABLE_COMPILE_CACHE
+#define DISABLE_COMPILE_CACHE
```
Comment (LinkIsGrim) [inline addons/feedback/script_component.hpp]:
```suggestion
// #define DISABLE_COMPILE_CACHE
```

File: addons/feedback/functions/fnc_effectTearHaze.sqf [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+ * [true, 0.5] call kat_feedback_fnc_effectCoughing;
+ *
+ * Public: No
+ */
+
+params ["_enable", "_tear"];
+if (!_enable || _tear == 0) exitWith {
+    if (GVAR(tearHaze) != -1) then { GVAR(tearHaze) ppEffectEnable false; };
+};
+if (GVAR(tearHaze) != -1) then { GVAR(tearHaze) ppEffectEnable true; };
```
Comment (LinkIsGrim) [inline addons/feedback/functions/fnc_effectTearHaze.sqf:23]: I hate this. Don't have a better alternative, but I hate this.

File: addons/feedback/functions/fnc_effectTearHaze.sqf [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+GVAR(tearHaze) ppEffectAdjust [1.5,0.3,0.3,1.02,1.02,1.02,1.02,0.05,0.01,0.05,0.01,0.1,0.1,0.2,0.2];
```
Comment (LinkIsGrim) [inline addons/feedback/functions/fnc_effectTearHaze.sqf:25]: Holy magic numbers, batman

File: addons/chemical/script_component.hpp [hunk 8]
```diff
@@ -3,8 +3,8 @@
 #include "\x\kat\addons\main\script_mod.hpp"
 #include "\x\kat\addons\gui\gui_defines.hpp"
 
-// #define DEBUG_MODE_FULL
-// #define DISABLE_COMPILE_CACHE
+//#define DEBUG_MODE_FULL
+#define DISABLE_COMPILE_CACHE
```
Comment (LinkIsGrim) [inline addons/chemical/script_component.hpp]:
```suggestion
// #define DISABLE_COMPILE_CACHE
```

**Review by LinkIsGrim — APPROVED** [review]
lgtm otherwise

**Pull request merged**